### DISCUSSION TITLE
apt buildpack

### DISCUSSION THREAD

**Discussion by javierav** [paketo-buildpacks/feedback#195] [General]
Sometimes we need to add system dependencies like `libvips` for a Rails app, so we need a mechanism to install them inside the build and run images without to create a custom builder or stack. Heroku has something similar: https://github.com/heroku/heroku-buildpack-apt

**Comment by dmikusa** [comment]
You can use https://github.com/fagiani/apt-buildpack

There is also work coming in Buildpacks to make this natively possible. The workstream is called Extensions support. It right now allows you to make extensions that modify the build image dynamically, but not the run image. Run image is coming in the next phase of the implementation, hopefully not that far off. When that support has landed, Paketo will be talking more about how users can take advantage of it for this specific use case.

**Comment by javierav** [comment]
Thank you @dmikusa for your response.

I had already seen that repository with the buildpack, but it has two implications:

1. I have to use my own builder instead of using the one provided by Paketo.
2. From what I've seen it implements the buildpacks.io API 0.4 which is deprecated. https://github.com/fagiani/apt-buildpack/issues/21

**Reply by dmikusa** [reply]
If you are using `pack build` you can use the `-b` argument to override the builder-provided list of buildpacks and include additional buildpacks. Just keep in mind that `-b` completely overrides the list of buildpacks to use, so you need the complete list of what is required for your app, not just the apt buildpack.

For reference, this highlights other options as well. https://stackoverflow.com/questions/69447497/is-it-possible-to-customize-docker-image-generated-with-spring-native-with-buil/69466535#69466535

I would not worry about the API being 0.4. At this point, it's just deprecated so it is not strictly a problem.

**Reply by cldfzn** [reply]
Now that buildpacks <=0.6 have been dropped by the latest lifecycle, is the only current option to maintain a run image?

**Reply by dmikusa** [reply]
It's unfortunate that buildpack isn't being updated as it is still useful. I've forked it and published it under a new buildpack id, `dmikusa/apt-buildpack`, see https://github.com/dmikusa/apt-buildpack. You can give that a try.

**Reply by cldfzn** [reply]
Thank you so much for doing that, that's awesome. I'll give it a try in a bit.

**Reply by dmikusa** [reply]
If you hit any problems, open an issue on that repo. I'll respond over there.
